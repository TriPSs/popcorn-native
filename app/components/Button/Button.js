import React from 'react'
import PropTypes from 'prop-types'
import { StyleSheet } from 'react-native'

import colors from 'modules/colors'

import TextButton from '../TextButton'

const defaultPadding = {
  paddingLeft  : 16,
  paddingRight : 16,
  paddingTop   : 8,
  paddingBottom: 8,
  borderRadius : 4,
}

export const styles = StyleSheet.create({

  default: {
    ...defaultPadding,

    backgroundColor: colors.BUTTON_DEFAULT,
  },

  primary: {
    ...defaultPadding,

    backgroundColor: colors.BUTTON_PRIMARY,
  },

})

export const Button = ({ children, onPress, variant, style, ...rest }) => (
  <TextButton
    onPress={onPress}
    style={{
      ...styles[variant],
      ...style,
    }}
    fontWeight={'bold'}
    {...rest}>
    {children}
  </TextButton>
)

Button.propTypes = {
  onPress : PropTypes.func.isRequired,
  children: PropTypes.string.isRequired,
  variant : PropTypes.oneOf(['default', 'primary']),
}

Button.defaultProps = {
  variant: 'default',
  style  : {},
}

export default Button
