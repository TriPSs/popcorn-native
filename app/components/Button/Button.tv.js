import React, { useState } from 'react'
import PropTypes from 'prop-types'
import { StyleSheet, TouchableHighlight } from 'react-native'

import colors from 'modules/colors'
import dimensions from 'modules/dimensions'

import BaseButton from '../BaseButton'

const defaultPadding = {
  paddingLeft: dimensions.UNIT * 2,
  paddingRight: dimensions.UNIT * 2,
  paddingTop: dimensions.UNIT,
  paddingBottom: dimensions.UNIT,
  borderRadius: dimensions.BORDER_RADIUS,
}

export const styles = StyleSheet.create({

  default: {
    ...defaultPadding,
  },

  primary: {
    ...defaultPadding,
    backgroundColor: colors.BUTTON_PRIMARY
  },

  active: {

  },

  focused: {
    // backgroundColor: 'pink'
  }
})

export const Button = ({ children, variant, style, onFocus, ...rest }) => {
  const [focus, toggleFocus] = useState(false)

  return (
    <BaseButton
      style={{
        ...styles[variant],
        ...style,
        ...focus ? styles.focused : {}
      }}
      component={TouchableHighlight}
      onFocus={() => {
        if (onFocus) {
          onFocus()
        }
        console.log('button onFocus')
        toggleFocus(true)
      }}
      onBlur={() => {
        console.log('button onBlur')
        toggleFocus(false)
      }}
      fontWeight={'bold'}
      {...rest}>
      {children}
    </BaseButton>
  )
}

Button.propTypes = {
  onPress: PropTypes.func.isRequired,
  onFocus: PropTypes.func,
  onBlur: PropTypes.func,
//  children: PropTypes.string.isRequired,
  variant: PropTypes.oneOf(['default', 'primary']),
}

Button.defaultProps = {
  variant: 'default',
  onFocus: null,
  onBlur: null,
  style: {},
}

export default Button
