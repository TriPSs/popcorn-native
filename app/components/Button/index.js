import useCorrect from 'modules/useCorrect'

import ButtonMobile from './Button'
import ButtonTV from './Button.tv'

export default useCorrect(ButtonMobile, null, ButtonTV)
