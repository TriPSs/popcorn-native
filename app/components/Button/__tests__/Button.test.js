import React from 'react'
import { shallow } from 'enzyme'

import Button from '../Button'

describe('(Component) Button', () => {

  it('should render default', () => {
    const result = shallow(
      <Button
        onPress={jest.fn()}>
        test
      </Button>,
    )

    expect(result).toMatchSnapshot()
  })

  it('should render primary', () => {
    const result = shallow(
      <Button
        onPress={jest.fn()}
        variant={'primary'}>
        test
      </Button>,
    )

    expect(result).toMatchSnapshot()
  })

})
