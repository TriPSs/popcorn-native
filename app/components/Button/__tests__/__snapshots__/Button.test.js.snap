// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Button should render default 1`] = `
<TextButton
  color="white"
  emphasis="high"
  fontWeight="bold"
  onPress={[MockFunction]}
  style={
    Object {
      "borderRadius": 5,
      "paddingBottom": 8,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 8,
    }
  }
  upperCase={true}
  variant="button"
>
  test
</TextButton>
`;

exports[`(Component) Button should render primary 1`] = `
<TextButton
  color="white"
  emphasis="high"
  fontWeight="bold"
  onPress={[MockFunction]}
  style={
    Object {
      "backgroundColor": "#FB9B42",
      "borderRadius": 5,
      "paddingBottom": 8,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 8,
    }
  }
  upperCase={true}
  variant="button"
>
  test
</TextButton>
`;
